### DISCUSSION TITLE
How to select a specific version of a tool in a nuget package

### DISCUSSION THREAD

**Discussion by StillBitterAboutAerith** [cake-build/cake#4154] [General]
Greetings all!
I apologize in advance, I am a little new to Cake and am not completely sure how to ask this question so I will do my best. 

I am currently dealing with an issue. I am in the process of trying to add a tool to a cake based build library via Nuget. The nuget package I am pulling in has a number of build targets shipped for a few different versions of .net. The library itself is built in .Net 6.0 but Cake seems to be registering a different version of the tool. Is there a way to force registering based on subdirectory inside of the package or something? I tried adding an "include" parameter to the URI when registering the tool but it does not seem to do anything.

**Comment by augustoproiete** [comment]
Hi, which tool provider are you using? `nuget` or `dotnet`? More info: https://cakebuild.net/docs/writing-builds/preprocessor-directives/tool/

If you can provide an example of your build script and the package being referenced, it should be easy to troubleshoot.
